cpp/src/umap/simpl_set_embed/optimize_batch_kernel.cuh

```diff
@@ -315,8 +313,6 @@ CUML_KERNEL void optimize_batch_kernel(T const* head_embedding,
       auto grad_d = T(0.0);
       if (repulsive_grad_coeff > T(0.0))
         grad_d = clip<T>(repulsive_grad_coeff * (current[d] - negative_sample[d]), T(-4.0), T(4.0));
-      else
```

Comment:
I'm still curious about this change- I want to make sure we're not sacrificing the numerical stability by changing this based on limited evidence. What is the reason for changing this?

Reply:
This is a clear mistake here.
Here is the reference implementation : https://github.com/lmcinnes/umap/blob/a012b9d8751d98b94935ca21f278a54b3c3e1b7f/umap/layouts.py#L181

Reply:
@viclafargue I would not assume this is a mistake because it differs from the reference impl. The algorithms differ slightly in several ways. When we make these types of changes we should be validating and verifying them with tests. This is how more bugs can be introduced inadvertently.